JavaScript/chapter01/1.5 - OneAway/rroque98_sol.js

```diff
@@ -4,40 +4,65 @@ Given 2 strings write a function that will check if it is
 1 or 0 edits away*/
 
 const isOneAway = (str1, str2) => {
+  const diffInLen = Math.abs(str1.length - str2.length);
+  if (diffInLen > 1) {
+    return false;
+  }
+
+  let errorCount = 0;
   if (str1.length === str2.length) {
-    let errorCount = 0;
     for (let i = 0; i < str1.length; i++) {
       if (str1[i] !== str2[i]) {
         errorCount++;
-      }
-      if (errorCount > 1) {
-        return false;
+        if (errorCount > 1) {
+          return false;
+        }
       }
     }
   } else {
-    let errorCount = 0;
-    const longestStr = findLongestStr(str1, str2);
-    let x = 0;
-    for(let i = 0; i < longestStr; i++) {
-      if (str1[i] !== str2[x]) {
+    const longStr = str1.length > str2.length ? str1 : str2;
+    const shortStr = str1.length < str2.length ? str1 : str2;
+    for (
+      let i = 0, x = 0;
+      i < longStr.length && x < shortStr.length;
+      i++, x++
+    ) {
+      if (longStr[i] !== shortStr[x]) {
         errorCount++;
-        x++;
+        if (errorCount > 1) {
+          return false;
+        }
+        i++;
       }
-      if (errorCount > 1) {
-        return false;
-      }
-      x++;
-    }
-    // ***** Helper functions ********
-    function findLongestStr(str1, str2) {
-      str1 > str2 ? str1 : str2;
     }
   }
   return true;
-}
+};
 
 // TESTS
-console.log(isOneAway('pale', 'ple') === true);
-console.log(isOneAway('pales', 'pale') === true);
-console.log(isOneAway('pale', 'bale') === true);
-console.log(isOneAway('pale', 'bake') === false);
+console.log(isOneAway('pale', 'ple') === true); // deletion
```

Comment:
maybe try restructuring the tests like this:

```javascript
function runTests(cases, expected) {
  for (const [str1, str2] of cases) {
    console.log(isOneAway(str1, str2) === expected);
  }
}

runTests([
  ['pale', 'opale'], // insertion in beginning
  // ...
], true);
```

Reply:
Thanks for the feedback. Definitely helps make the code more DRY